### ISSUE TITLE
[enhancement]: Magic Prompts

### ISSUE THREAD

**Issue by xaroth8088** [invoke-ai/InvokeAI#6719]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What should this feature add?

The ability to use an LLM (e.g. LLama) to take in a short prompt and expound on it to enhance generated image quality.  

### Alternatives

Running something like GPT4All in parallel, or using ChatGPT/Claude/etc. to generate the enhanced prompts.

### Additional Content

ChatGPT/DALL-E 3 does this secretly in the background (though you can get at the actual prompt after the image is generated), and Ideogram has a toggle for this.

Natural follow-on features would include:

* Editing the LLM prompt template that's used to generate the enhanced prompt
* LLM model management for the prompt generator, to pull down any compatible model

**Comment by stripealipe** [comment]
Yeah, I use similar in some of my Comfy workflows and it's super cool for creative purposes.
For Forge and now Invoke I'm typing manually into LM Studio which is of course somewhat slow...

Invoke is great btw - thanks for all the hard work so far!

**Comment by dennisrall** [comment]
There are also some specialized models for this like [https://huggingface.co/succinctly/text2image-prompt-generator]()

**Comment by Pfannkuchensack** [comment]
#9152
